extension/registry/holixon/holixon-axon/src/main/kotlin/AxonAvroRegistry.kt

```diff
@@ -0,0 +1,75 @@
+package io.holixon.avro.adapter.registry.axon
+
+import io.holixon.avro.adapter.api.*
+import io.holixon.avro.adapter.registry.axon.api.*
+import org.apache.avro.Schema
+import org.axonframework.commandhandling.gateway.CommandGateway
+import org.axonframework.messaging.responsetypes.ResponseTypes
+import org.axonframework.messaging.responsetypes.ResponseTypes.instanceOf
+import org.axonframework.queryhandling.QueryGateway
+import org.reactivestreams.Publisher
+import java.util.*
+import kotlin.Result.Companion.failure
+import kotlin.Result.Companion.success
+
+/**
+ * Registry storing schemas in event store.
+ */
+open class AxonAvroRegistry(
+  private val commandGateway: CommandGateway,
+  private val schemaIdSupplier: SchemaIdSupplier,
+  private val schemaRevisionResolver: SchemaRevisionResolver,
+  private val queryGateway: QueryGateway
+) : AvroSchemaRegistry {
+
+  companion object {
+    val returnClazz = AvroSchemaWithId::class.java
+  }
+
+  override fun register(schema: Schema): AvroSchemaWithId {
+    val schemaId = schemaIdSupplier.apply(schema)
+    commandGateway.sendAndWait<Void>(
+      RegisterAvroSchemaCommand(
+        namespace = schema.namespace,
+        name = schema.name,
+        revision = schemaRevisionResolver.apply(schema).orElse(null),
+        schema = schema.toString(),
+        schemaId = schemaId
+      )
+    )
+
+    val query = queryGateway.subscriptionQuery(
```

Comment:
why bother about subcription queries, when (see other comment) there never will be a change for an aggregate with "fingerprint" id?